bcgov/HOUS-permit-portal
# add optimized user archive, restore, and user sign in tracking

## Description

Adds user archive ad restore
Refactors user search
Add User last sign in tracking

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-232?atlOrigin=eyJpIjoiMDllYWRhNGNiMDAyNDZjZmE5NDNhOGI5MmFlZDY2NWQiLCJwIjoiaiJ9

## Steps to QA

Log in as super admin, log in to jurisdiction, archive users from manage menu
These users will appear in the archived list (see by clicking button at the bottom)

Notice that there are no repeated search or fetchJurisdiction calls! 🔥 

## UI Accessibility Checklist

No new major UI changes

## General Checklist

- [ x] ✅ Provide tests for your changes where relevant
- [x ] 📝 Use descriptive commit messages
- [x ] 📗 Update any related documentation and include any relevant screenshots
- [x ] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size
- [x ] 🗂️ Move your relevant story/task to the _Ready for Code Review_ state
